### ISSUE TITLE
false positive on label-content-name-mismatch rule

### ISSUE THREAD

**Issue by devaradhanm** [dequelabs/axe-core#1635]
The rule seems to evaluate the innerText value & compares it with the aria-label value. This results in false positive, when we have icon as a child element that has some content, even though visually the user only sees icon instead of text.
eg where it fails:
```
<a id="link1" tabindex="0" href="./some-link" aria-label="Favorites">
  <div><span aria-hidden="true" class="some-icon"></span></div>
  <div><span aria-hidden="true">Favorites</span></div>
</a>
```
https://codepen.io/devaradhanm/pen/xoZdmd

including @iamrafan , for his inputs too.

**Comment by straker** [comment]
I'm not very familiar with this rule so am not sure what the purpose of it is. The [issue where it was proposed](https://github.com/dequelabs/axe-core/issues/1149) links out to a 404 page, so I'll have to get back to you on if this is a bug or works as intended.

**Comment by dylanb** [comment]
@straker the link in that issue is broken

**Comment by WilcoFiers** [comment]
IMO we should include that icon character. Things that can't be pronounced should be ignored for the purpose of checking if things match.

**Comment by devaradhanm** [comment]
@WilcoFiers , we also saw this issue in news.google.com. The left pane icons have readable text that gets rendered as icon.

![image](https://user-images.githubusercontent.com/29855291/60747252-d3171100-9f38-11e9-8cc2-df060dd36560.png)

![image](https://user-images.githubusercontent.com/29855291/60747268-f17d0c80-9f38-11e9-9945-b9adc75bd1e3.png)

**Comment by straker** [comment]
With the help of @mfairchild365 we were able to determine that emoji/unicode inside of a control with `aria-label` set to the visible non-emoji/unicode will allow [Dragon Naturally Speaking to activate the control](https://decorous-jacket.glitch.me/). With that in mind, we'll fix the rule to take them into account.

**Comment by jeankaplansky** [comment]
This goes in the release notes.

**Comment by chandana7393** [comment]
Tested, working as expected.
![label_content_name_mismatch](https://user-images.githubusercontent.com/45285854/66819049-e5eba980-ef5b-11e9-8e6e-cc013e20b717.PNG)
**Tested Environment:**
Axe-coconut - 4.0.0.28804v
Axe chrome - 4.0.0.28714v
Axe-Firefox - 4.0.0.28714v
Chrome -  77.0.3865.120v
Firefox - 69.0.3 v
OS - Windows 10 64 bit.
